api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -269,7 +273,12 @@ public DataSet saveDataSet(DataSetRequest dataSetRequest, Long userId) {
   @Override
   public DataSet saveDataSet(DbDataset dataset) {
     try {
-      return dataSetMapper.dbModelToClient(dataSetDao.save(dataset));
+      DbDataset savedDbDataSet = dataSetDao.save(dataset);
```

Comment:
i am using savedDbDataSet instead of original is because, saveDataSet method is also used for createDataSet and therefore there is a chance that dataset.getDataSetId() would be null

Reply:
oh nvm i re-read this.

Reply:
Done